docs/authentication.md

```diff
@@ -103,6 +103,30 @@ export default Profile
 
 An advantage of this pattern is preventing a flash of unauthenticated content before redirecting. It's important to note fetching user data in `getServerSideProps` will block rendering until the request to your authentication provider resolves. To prevent creating a bottleneck and increasing your TTFB ([Time to First Byte](https://web.dev/time-to-first-byte/)), you should ensure your authentication lookup is fast. Otherwise, consider [static generation](#authenticating-statically-generated-pages).
 
+## Authenticating API Routes
+
+API routes are typically authenticated using a secret stored in a cookie or delivered through the `Authorization` header.
+
+Authentication providers offer helpers to verify this secret and return authenticated user data to your API route.
+
+Let's look at an example that uses middleware to decorate the request object with user data.
+
+```js
+import { withSession } from '@clerk/nextjs/api'
+
+// Use withSession to decorate `req` with a session object
+export default withSession((req, res) => {
```

Comment:
Could we make this provider agnostic? We'd like to avoid picking favorites here 😄

Reply:
Yes, definitely.  In general, I tried to use the precedent from the `with-iron-session` examples above - talking generically about the pattern but leveraging a specific example.

I just added a commit that obscures the import:
https://github.com/vercel/next.js/pull/28906/commits/7a57cd445c8af25cb9ac7ce93d924bed788c5174

Reply:
It's still referencing Clerk below - I've tried to keep this provider agnostic, with iron session being an OSS library. Can we use that as the example and ensure the code snippet is reflected in the example?

Reply:
e.g. here https://github.com/vercel/next.js/blob/canary/examples/with-iron-session/pages/api/user.js

Reply:
Ok, just pushed an update with the linked example and re-ran linters.

Lmk if there are any other issues!